apps/client/src/app/services/internet-identity.service.ts

```diff
@@ -0,0 +1,42 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { AuthClient } from '@dfinity/auth-client';
+import { OAuthResponse } from '@ghostfolio/common/interfaces';
+import { EMPTY } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class InternetIdentityService {
+  public constructor(private http: HttpClient) {}
+
+  public async login(): Promise<OAuthResponse> {
+    const authClient = await AuthClient.create();
+
+    return await new Promise((resolve, reject) => {
+      authClient.login({
+        onError: async () => {
+          return reject();
+        },
+        onSuccess: () => {
+          const principalId = authClient.getIdentity().getPrincipal();
+
+          this.http
+            .get<OAuthResponse>(
+              `/api/v1/auth/internet-identity/${principalId.toText()}`
+            )
+            .pipe(
```

Comment:
you don't unsubscribe here so there is a possible memory leak

`.pipe(first(), ...` or `takeUntil` etc.

Reply:
Oh, I did not know that it's also possible in services with `providedIn: 'root'` 🙂